src/llmcompressor/modifiers/awq/base.py

```diff
@@ -96,53 +112,138 @@ class AWQModifier(Modifier):
     duo_scaling: bool = True
 
-    _resolved_mappings: List[ResolvedMapping] = []
-    _scales: Dict[str, Union[torch.Tensor, List[torch.Tensor]]] = {}
-    _module_kwargs: Dict = {}
+    # Private vars set during validation
+    _num_bits: Optional[int] = PrivateAttr(default=None)
+    _symmetric: Optional[bool] = PrivateAttr(default=None)
+    _group_size: Optional[int] = PrivateAttr(default=None)
+
+    # Private vars set during initialization, cleared during finalization
+    _resolved_mappings: List[ResolvedMapping] = PrivateAttr(default_factory=list)
+    _activations: Dict[str, Union[torch.Tensor, List[torch.Tensor]]] = PrivateAttr(
+        default_factory=dict
+    )
+    _activation_hooks: Set[RemovableHandle] = PrivateAttr(default_factory=set)
+    _module_kwargs: Dict = PrivateAttr(default_factory=dict)
+
+    @model_validator(mode="after")
+    def validate_model_after(model: "AWQModifier") -> "AWQModifier":
+        if not model.config_groups and not model.scheme:
+            raise ValueError("AWQ requires either a config_groups or a scheme")
+
+        # TODO better way to do this?
+        config_groups = model.config_groups or preset_name_to_scheme(model.scheme)
+
+        num_bits_set = set(
+            group.weights.num_bits
+            for group in config_groups.values()
+            if group.weights is not None
+        )
+        assert (
+            len(num_bits_set) == 1
+        ), "In AWQ, all config groups must use the same configuration for num_bits"
+
+        model._num_bits = next(iter(num_bits_set))
+
+        symmetric_set = set(
+            group.weights.symmetric
+            for group in config_groups.values()
+            if group.weights is not None
+        )
+        assert (
+            len(symmetric_set) == 1
+        ), "In AWQ, all config groups must use the same configuration for symmetric"
+
+        model._symmetric = next(iter(symmetric_set))
+
+        group_size_set = set(
+            group.weights.group_size
+            for group in config_groups.values()
+            if group.weights is not None
+        )
+        assert (
+            len(group_size_set) == 1
+        ), "In AWQ, all config groups must use the same configuration for group_size"
+
+        model._group_size = next(iter(group_size_set))
+
+        return model
 
     def on_initialize(self, state: State, **kwargs) -> bool:
         """
-        Initialize and run AWQ on the given state
+        Initialize AWQ on the given state
 
         :param state: state to run AWQ on
         :return: True on a successful run, False otherwise
         """
 
+        # apply config to model and prepare calibration hooks
+        if QuantizationMixin.has_config(self):
+            QuantizationMixin.initialize_quantization(self, state.model)
+
         self._set_resolved_mappings(state.model)
 
         with calibration_forward_context(state.model):
             self._set_module_kwargs(state.model, state.data.calib)
```

Comment:
Can we get rid of this?

Reply:
I'm hoping so, but we need it to run forward passes during smoothing. Bit different than GPTQ, AWQ needs the full output as well as the quantized output for the loss function. Maybe we can just cache output activations as well?

Reply:
Unfortunately, we need this for forward passes through the parent modules, which are not always Linear layers. So let's leave as is for now, and we can tackle this with more hooks to inputs to each of the smoothing layers in a follow-up?